### PR TITLE
add java example

### PR DESCRIPTION
**Pull request by dt**
note java requires that the class name and file name match, thus the renamed file.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/142)

### CODE REVIEW

File: _includes/app/BasicSample.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import java.sql.*;
+
+/*
+You can compile and run this example with a command like:
+  javac BasicSample.java && java -cp .:~/path/to/postgresql-9.4.1208.jar BasicSample
+You can download the postgres JDBC driver jar from https://jdbc.postgresql.org.
+*/
+public class BasicSample {
+    public static void main(String[] args) throws ClassNotFoundException, SQLException {
+		// Load the postgres JDBC driver.
```
Comment (spencerkimball) [inline _includes/app/BasicSample.java]: What's up with the strange white space?

Comment (dt) [inline _includes/app/BasicSample.java]: weird. fixed.

**Comment by jseldess** [comment]
Thanks, @dt. Would you mind adding comments like we have in the [node](https://github.com/cockroachdb/docs/blob/gh-pages/_includes/app/basic-sample.js) and [python](https://github.com/cockroachdb/docs/blob/gh-pages/_includes/app/basic-sample.py) samples? And who else needs to review this?

File: _includes/app/BasicSample.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import java.sql.*;
```
Comment (JackKrupansky) [inline _includes/app/BasicSample.java]: Any copyright notice and license needed?

File: _includes/app/BasicSample.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+/*
+You can compile and run this example with a command like:
```
Comment (JackKrupansky) [inline _includes/app/BasicSample.java]: I'd recommend the Apache projects' style for multi-line /* comments - with a space and asterisk and another space at start of each subsequent line.

For example, see:
https://github.com/apache/lucene-solr/blob/master/lucene/analysis/common/src/java/org/apache/lucene/analysis/charfilter/BaseCharFilter.java

File: _includes/app/BasicSample.java (hunk 2 above)
Comment (dt) [inline _includes/app/BasicSample.java]: These are just snippets which will be embedded in other docs pages.

File: _includes/app/BasicSample.java (hunk 3 above)
Comment (dt) [inline _includes/app/BasicSample.java]: That format is specific to documentation comments (i.e. javadoc), and I don't think I'd call this a javadoc comment.

**Comment by andreimatei** [comment]
:lgtm: 

---

Review status: 0 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[_includes/app/BasicSample.java, line 32 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/142#-KDeqM2XZTm3L3svHhcQ:-KDeqM2Y-IV2mvZQR0VE:794365359) ([raw file](https://github.com/cockroachdb/docs/blob/f4280e0535da663324cf3c8bb08477d70ebbb5d2/_includes/app/BasicSample.java#L32)):*
no transactions

---

*[_includes/app/BasicSample.java, line 48 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/142#-KDerVP60nmBpkLdaAQ4:-KDerVP60nmBpkLdaAQ5:-2071447779) ([raw file](https://github.com/cockroachdb/docs/blob/f4280e0535da663324cf3c8bb08477d70ebbb5d2/_includes/app/BasicSample.java#L48)):*
no more BigDecimal?

---

*[_includes/app/BasicSample.java, line 51 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/142#-KDerZw9bfoHnuk59Efh:-KDerZw9bfoHnuk59Efi:1166149827) ([raw file](https://github.com/cockroachdb/docs/blob/f4280e0535da663324cf3c8bb08477d70ebbb5d2/_includes/app/BasicSample.java#L51)):*
do we need a try - finally ?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/docs/142#-:-KDerc95DVUk3xrGkBYY:980077423)*

File: _includes/app/BasicSample.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            db.createStatement().execute("INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 230)");
+
+            // Print out the starting balances.
+            System.out.println("Before transaction:");
+            ResultSet res = db.createStatement().executeQuery("SELECT id, balance FROM accounts");
+            while (res.next()) {
+                System.out.printf("\taccount %s: %s\n", res.getInt("id"), res.getInt("balance"));
+            }
+
+            // Transfer $100 from account 1 to account 2.
```
Comment (jseldess) [inline _includes/app/BasicSample.java]: Can you remove $ for this and the next comment?

Comment (dt) [inline _includes/app/BasicSample.java]: whoops, fixed, thanks.

**Comment by jseldess** [comment]
@dt, the [basic python](https://github.com/cockroachdb/docs/blob/gh-pages/_includes/app/basic-sample.py) and [ruby](https://github.com/cockroachdb/docs/blob/gh-pages/_includes/app/basic-sample.rb) samples are now up-to-date; they no longer include the transfer transaction. 

Would you mind simplifying this sample one more time to mirror what we're doing in those others? Then, once Andrei's go transaction sample is merged, you can open another PR to mirror that in java. Thanks!

File: _includes/app/BasicSample.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+You can compile and run this example with a command like:
+  javac BasicSample.java && java -cp .:~/path/to/postgresql-9.4.1208.jar BasicSample
+You can download the postgres JDBC driver jar from https://jdbc.postgresql.org.
+*/
+public class BasicSample {
+    public static void main(String[] args) throws ClassNotFoundException, SQLException {
+        // Load the postgres JDBC driver.
+        Class.forName("org.postgresql.Driver");
+
+        // Connect to the "bank" DB on a local cockroach node.
```
Comment (jseldess) [inline _includes/app/BasicSample.java]: Wouldn't you leave the bank db out of the connection call since it hasn't been created yet?

Comment (dt) [inline _includes/app/BasicSample.java]: eh, then you just have to set it later. cockroach does not require a db exist at connection time, only when you use a table in it.

**Comment by jseldess** [comment]
OK, after chatting offline, I now understand the value of including the db in the connection request rather than using `SET DATABASE`. LGTM.

**Comment by jseldess** [comment]
Reviewed 2 of 3 files at r1, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/docs/142)*

**Pull request merged**